staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -569,6 +571,83 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	return nil
 }
 
+// EventHandlerRemovable is an extension interface
+// for a SharedIndexInformer that allows to remove event handlers again.
+// It is possible to remove only event handlers that are (go) comparable,
+// This means event handlers like ResourceEventHandlerFuncs cannot be removed
+// if added by value and not by reference, because they contain function pointers
+// that are not comparable in go. Trying to remove an uncomparable handler
+// results in an appropriate error.
+// This feature can be used to dynamically add and remove watches again depending
+// on the actual state of a controller.
+type EventHandlerRemovable interface {
+	RemoveEventHandler(handler ResourceEventHandler) error
+}
+
+// ExtendedSharedIndexInformer is an extended interface to the
+// SharedIndexInformer informer interface implemented by the standard
+// sharedIndexInformer implementation,
+// It offers the removing of handlers and some state info useful
+// to use after handler removal to decide whether an informer
+// is still required.
+type ExtendedSharedIndexInformer interface {
```

Comment:
I did not know whether the informer interface is used by other implementations, also. In such a case extending the interface would break all those other implementations.

This was the reason I decided not to extend the existing interface, but introduce ab extended one.

If you don't see a compatibility problem here, I surely can extend the existing interface, instead,

Reply:
I know in kubernetes/kubernetes there are no other implementations, not sure about outside of k/k. I'll leave it @yliaog or someone from sig-api-machinery to decide.

Reply:
it is a breaking change if extending the SharedIndexInformer Interface directly. For breaking change, a few options: 1) send an email to kubernetes-sig-api-machinery@google.com, or kubernetes-dev@google.com 2) discuss it in biweekly sig-api-machinery meeting. 3) add to release notes about the breaking change

I think it is cleaner to extend the SharedIndexInformer directly, but it's better to check the community to ensure it won't break people unintentionally.

Reply:
I've sent I mail to  kubernetes-sig-api-machinery@google.com referring to this PR

Reply:
I'll put it on the agenda; I can see either breaking compat or adding a "EventHandlerRemovable" as above, but I can't see making this "ExtendedSharedIndexInformer" interface; what will we do when we add another method, make a "ExtendedExtendedSharedIndexInformer"?